nats.go

```diff
@@ -287,6 +287,9 @@ type Options struct {
 	// Token sets the token to be used when connecting to a server.
 	Token string
 
+	// TokenFunc returns a token to be used when connecting to a server.
+	TokenFunc func() string
```

Comment:
Should we create a type for `func() string` to be consistent with the Callbacks?